docs/data/routes/docs/nextjs/troubleshooting/en.md

```diff
@@ -0,0 +1,156 @@
+      value :"true"
+      }
+    ]
+  })
+  { 
+    total
+    results {
+        url {
+          path
+        }
+      }
+  }
+}
+```
+
+If you do not see expected results, try the following steps:
+
+1. Populate Solr Schema (Sitecore Control Panel > “Populate Solr Managed Schema”)
+
+2. Rebuild indexes (Sitecore Control Panel > “Indexing Manager”)
+
+## Errors regarding SSL certificates
+
+Error examples
+> Error: unable to verify the first certificate
+>
+> Error: UNABLE_TO_VERIFY_LEAF_SIGNATURE
+
+> UnauthorizedError: invalid signature
+
+If you are working locally, make sure that you have [followed the steps](todo://add_link_to_new_page) for enabling self-signed certificates in Node. Check that you have the Node Extra Certificates path in your system variables.
+
+Node that the steps are different, depending on whether you're using a containerized or on-prem installation of Sitecore.
+
+## Errors when generating GraphQL introspection data
+
+This task is performed if the GraphQL schema changes. It depends on the `scjssconfig.json` file being present and populated.
+
+## Errors deploying a JSS app locally
+
+```
+IMPORT ERROR(S) OCCURRED!
+Exception thrown while importing JSS app
+Exception: Sitecore.Exceptions.AccessDeniedException
+Message: AddFromTemplate - Add access required
+```
+
+This occurs when the import role doesn't have the correct permissions. See the following Knowledge Base article for the solution: https://kb.sitecore.net/articles/650791
+
+## General solution checklist
+
+- Verify that all environment variables defined in the [`.env` file](https://github.com/Sitecore/jss/blob/master/samples/nextjs/.env) file are provided. On local environments, an `.env.local` file should be for this purpose.
+
+- Check that Sitecore has an item for your API key  in the `/sitecore/system/Settings/Services/API Keys` folder and that this item's fields are populated.
+
+- Verify that the "JSS Editing Secret" is set. The`JSS_EDITING_SECRET` [value used by your Next.js app](https://github.com/Sitecore/jss/blob/cb32d3a21b87f488bd4bb5d311d556fd1f8354c4/samples/nextjs/.env#L18) should match the [value used by your Sitecore instance](https://github.com/Sitecore/jss/blob/cb32d3a21b87f488bd4bb5d311d556fd1f8354c4/samples/nextjs/sitecore/config/JssNextWeb.config#L37).
+
+- Verify that that GraphQL endpoint is correct in [`package.json`](https://github.com/Sitecore/jss/blob/cb32d3a21b87f488bd4bb5d311d556fd1f8354c4/samples/nextjs/package.json#L12) and in the Sitecore [app definition](https://github.com/Sitecore/jss/blob/cb32d3a21b87f488bd4bb5d311d556fd1f8354c4/samples/nextjs/sitecore/config/JssNextWeb.config#L72). Try opening the GraphiQL interface in a browser, using `<sitecore hostname + graphQL endpoint>/ui?sc_apikey=<api key>`, to make sure the endpoint is working
+
+- If using [ngrok](https://ngrok.com/) for any part of your local workflow, verify that you are using the `host-header` flag.
+For example,
+```
+ngrok http -host-header=rewrite 3000
```

Comment:
I used it without `host-header`, is it required? You should use host header when you want tunnel https instance
for example: 
`ngrok http -host-header="jsssite" https://jsssite

Reply:
Good point. I know it's at least required for [enabling tracking and analytics w/ SSR](https://jss.sitecore.com/docs/nextjs/tracking-and-analytics/configuration#secure-cookies). I think it would also be required if you've configured specific CORS origins on your API key.

Reply:
I think getting into the weeds of ngrok usage is beyond what this page should cover. But I think it's safe to say that if nrok *is* being used, and it's not working as expected, the host-header option is a good thing to try.